libraries/chain/include/eosio/chain/webassembly/eos-vm-oc/code_cache.hpp

```diff
@@ -37,6 +37,12 @@ using allocator_t = bip::rbtree_best_fit<bip::null_mutex_family, bip::offset_ptr
 
 struct config;
 
+// get_descriptor_for_code failure reasons
+enum class get_cd_failure {
+   temporary, // oc compile not done yet, users like read-only trxs can retry
+   permanent  // oc will not start, users should not retry
+};
```

Comment:
This can be moved into `code_cache_base` class I think and then it could be `code_cache_base::cd_failure` which seems more descriptive and easier to read/understand.